### PR TITLE
adjust android / linux build config for cocos CLI

### PR DESCRIPTION
**Pull request by drelaptop**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "tools/cocos2d-console"]
         path = tools/cocos2d-console
-        url = git://github.com/cocos2d/cocos2d-console.git
```
Comment (minggo) [inline .gitmodules]: Should not change it.

**Comment by crazyhappygame** [comment]
@drelaptop why did you set ndk-build as default instead of cmake? Do we have any problems with cmake?

**Comment by drelaptop** [comment]
@crazyhappygame Not found any problems with cmake yet, but ndk-build is used for a long time in cocos2d-x, it's more stable and well-tested than cmake for android build. so set it as the default.

**Comment by crazyhappygame** [comment]
@drelaptop We also used gcc and "GNU's libstdc++" as default for long time but that was changed few moths ago :)
Please see https://developer.android.com/ndk/guides/index.html
> Android Studio's default build tool to compile native libraries is CMake

I think that after switch from gcc to clang and from"GNU's libstdc++" to libc++ we should complete transition to latest build/toolset and also switch to CMake as default for Android.
Additionally for CMake we have prebuild libraries.

@mars3142, @mozartalouis is there any reason to use ndk-build instead of CMake for new projects?

**Comment by crazyhappygame** [comment]
@drelaptop @minggo do we have release date for cocos2d-x.3.17?
There are some thinks to do before release e.g.: https://discuss.cocos2d-x.org/t/important-targetsdk-and-ndk-changes-in-android-app-publishing/40621

**Comment by drelaptop** [comment]
@crazyhappygame 3.17 release is on the way, the round 2 test will be coming at next week. As forecast we can release it in a month. 

About the android project configs, ABI have set `armeabi-v7a` as default,  versions will make the default :
```
PROP_COMPILE_SDK_VERSION=27
PROP_MIN_SDK_VERSION=14
PROP_TARGET_SDK_VERSION=27
```

File: .gitmodules (hunk 1 above)
Comment (drelaptop) [reply .gitmodules]: thanks for mention it, I will revert this file after corresponding https://github.com/cocos2d/cocos2d-console/pull/438 merged.

**Comment by crazyhappygame** [comment]
@drelaptop Thank you for update  android project configs.
But what about CMake?
I propose to set CMake as default in cocos2d-x.3.17 and in next release remove ndk-build support and Android.mk, Application.mk files.

**Comment by crazyhappygame** [comment]
@drelaptop I checked now ( and few months ago) "old prebuilds" for android does not work. But we have working android prebuilds for CMake.
Guys please make CMake as default for Android.  

> cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a:x86
> ....
17:00:05 Traceback (most recent call last):
17:00:05   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\bin\/cocos.py", line 1072, in <module>
17:00:05     run_plugin(command, argv, plugins)
17:00:05   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\bin\/cocos.py", line 961, in run_plugin
17:00:05     plugin.run(argv, dependencies_objects)
17:00:05   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_generate\gen_libs.py", line 79, in run
17:00:05     self.compile()
17:00:05   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_generate\gen_libs.py", line 176, in compile
17:00:05     self.compile_android()
17:00:05   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_generate\gen_libs.py", line 368, in compile_android
17:00:05     cocos.copy_files_with_config(copy_cfg, obj_dir, android_out_dir)
17:00:05   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\bin\cocos.py", line 799, in copy_files_with_config
17:00:05     src_dir, src_dir, dst_dir, include_rules, exclude_rules)
17:00:05   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\bin\cocos.py", line 818, in copy_files_with_rules
17:00:05     for name in os.listdir(src):
17:00:05 WindowsError: [Error 3] The system cannot find the path specified: 'g:\\j\\A\\external\\cocos2d-x\\tests/cpp-empty-test\\proj.android/app/build/intermediates/ndkBuild/release/obj/local/armeabi-v7a:x86/*.*'

**Comment by drelaptop** [comment]
@crazyhappygame  Keep the `ndk-build` as default is a prudent choice, @minggo also suggest me to do it, and the cmake-build isn't so perfect, for example `CocosPickLibs.cmake` is too complex, this should be improved in the future.

About the `gen-libs` error, I will check it. thanks for mention it. 

fixed at https://github.com/cocos2d/cocos2d-console/pull/439

**Comment by crazyhappygame** [comment]
gen-lib - I am looking forward to test it :)

**Comment by drelaptop** [comment]
module `cocos2d-console` have changed to `cocos2d`, and update to latest v3 https://github.com/cocos2d/cocos2d-console/commit/1b69aa81a05d727cd15d5d41c94c0e7332bb622e ,please merge @minggo

**Comment by crazyhappygame** [comment]
@drelaptop gen-libs does not work for me. For latest commit I have following error:
Could you check it:
```
20:09:58 g:\j\A>cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a:x86 
20:09:58 running: 'g:\j\A\external\cocos2d-x\tools/cocos2d-console/bin\cocos.bat compile -s g:\j\A\external\cocos2d-x\tests/cpp-empty-test -p android --mode release --app-abi armeabi-v7a:x86 --ap android-15'
20:09:58 
20:10:00 Building mode: release
20:10:00 Using Android Studio project : g:\j\A\external\cocos2d-x\tests\cpp-empty-test\proj.android
20:10:00 Building apk...
20:10:00 Please input the absolute/relative path of '.keystore' file:
20:10:00 Traceback (most recent call last):
20:10:00   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\bin\/cocos.py", line 1072, in <module>
20:10:00     run_plugin(command, argv, plugins)
20:10:00   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\bin\/cocos.py", line 961, in run_plugin
20:10:00     plugin.run(argv, dependencies_objects)
20:10:00   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_compile\project_compile.py", line 1519, in run
20:10:00     self.build_android()
20:10:00   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_compile\project_compile.py", line 537, in build_android
20:10:00     self.apk_path = builder.do_build_apk(build_mode, self._no_apk, output_dir, self._custom_step_args, self._ap, self)
20:10:00   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_compile\build_android.py", line 555, in do_build_apk
20:10:00     self._gather_sign_info()
20:10:00   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_compile\build_android.py", line 587, in _gather_sign_info
20:10:00     inputed = self._get_user_input(MultiLanguage.get_string('COMPILE_TIP_INPUT_KEYSTORE'))
20:10:00   File "g:\j\A\external\cocos2d-x\tools\cocos2d-console\plugins\plugin_compile\build_android.py", line 617, in _get_user_input
20:10:00     ret = raw_input()
20:10:00 EOFError: EOF when reading a line
20:10:00 Error running command, return code: 1.

```

**Comment by madebymozart** [comment]
@crazyhappygame Where is your `.keystore` file located?

`20:10:00 Please input the absolute/relative path of '.keystore' file:`

It seems like this is why it's failing

**Comment by crazyhappygame** [comment]
@mozartalouis Yes .keystore is problem. Question is why it does not work out of the box?
I would like to create prebuilt version of cocos2d-x. In prev releases this commands just worked:
`cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a:x86 `

Why I have to specify '.keystore' right now to create prebuilt? If for prebuilds project "cocos2d-x\tests\cpp-empty-test" is used than probably .keystore should be set properly and committed for this project?

**Pull request merged**